scripts/check_e2e_tests_are_captured_in_ci.py

```diff
@@ -0,0 +1,158 @@
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""A script to check that travis.yml file & protractor.conf.js have the
+same test suites."""
+
+from __future__ import absolute_import  # pylint: disable=import-only-modules
+from __future__ import unicode_literals  # pylint: disable=import-only-modules
+
+import os
+import re
+
+import python_utils
+import utils
+
+
+def get_e2e_suite_names_from_jobs_travis_yml_file():
+    """Extracts the env/jobs section from the .travis.yml file.
+
+    Returns:
+        list(str). A list of names of test suites in jobs section in travis.yml file.
+    """
+    travis_file = read_and_parse_travis_yml_file()
+    jobs_raw = travis_file['env']['jobs']
+    suites_from_jobs = []
+    # The following line extracts the test suite name from the jobs section that
+    # is in the form RUN_E2E_TESTS_ACCESSIBILITY=true.
+    test_regex = re.compile(r'(?<=RUN_E2E_TESTS_)(.*)(?=\=)')
+    for job in jobs_raw:
+        matches = test_regex.finditer(job)
+        for match in matches:
+            suites_from_jobs.append(
+                utils.snake_case_to_camel_case(match.group().lower()))
+
+    return sorted(suites_from_jobs)
+
+
+def get_e2e_suite_names_from_script_travis_yml_file():
+    """Extracts the script section from the .travis.yml file.
+
+    Returns:
+        list(str). A list of names of test suites in script section in travis.yml file.
+    """
+    travis_file = read_and_parse_travis_yml_file()
+    script_raw = travis_file['script']
+    # The following line extracts the test suites from patterns like
+    # bash scripts/run_e2e_tests.sh --suite="accessibility"
+    hyphen_between_regex = re.compile(
+        r'(?<= bash scripts/run_e2e_tests.sh --suite=")(.*)(?=")')
+    suites_from_script = []
+
+    for script in script_raw:
+        matches = hyphen_between_regex.finditer(script)
+        for match in matches:
+            suites_from_script.append(match.group())
+
+    return sorted(suites_from_script)
+
+
+def get_e2e_suite_names_from_protractor_file():
+    """Extracts the test suites section from the .travis.yml file.
+
+    Returns:
+        list(str). A list of all test suites in protractor.conf.js file.
+    """
+    protractor_config_file = read_protractor_conf_file()
+    # The following line extracts suite object from protractor.conf.js.
+    suite_object_string = re.compile(
+        r'suites = {([^}]+)}').findall(protractor_config_file)[0]
+
+    # The following line extracts the keys/test suites from the key: value pair
+    # from the suites object.
+    key_regex = re.compile(r'\b(.*?)(?=:)')
```

Comment:
In general, try to use a whitelisted charset (like [A-Za-z-_]) rather than `.*`. Have your regex be as specific as possible -- it's cleaner, and conveys intention more clearly.

Reply:
Done